extension.js

```diff
@@ -220,9 +222,26 @@ const GithubNotifications = new Lang.Class({
   },
 
   updateNotifications: function(data) {
+    let lastNotificationsCount = this.notifications.length;
+
     this.notifications = data;
     this.label && this.label.set_text('' + data.length);
     this.checkVisibility();
+    this.alertWithNotifications(lastNotificationsCount);
+  },
+
+  alertWithNotifications: function(lastCount) {
+    let newCount = this.notifications.length;
+
+    if (newCount && newCount != lastCount && this.showAlertNotification) {
```

Comment:
So from my testing the first thing I noticed was that we get notified even if we're actively deleting stuff. If I have 2 notifications and I'm reading a PR (bringing it down to 1) I'll get notified.

What would you think about having something like that?
```javascript
if (newCount && newCount > lastCount && this.showAlertNotification) {
```

Reply:
That's a great point. Let me change that =)